drivers/ng_at86rf2xx/ng_at86rf2xx.c

```diff
@@ -0,0 +1,256 @@
+    tmp |= (NG_AT86RF2XX_TRX_CTRL_2_MASK__SUB_MODE
+            | NG_AT86RF2XX_TRX_CTRL_2_MASK__OQPSK_SCRAM_EN);
+#endif
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__TRX_CTRL_2, tmp);
+    /* enable interrupts */
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__IRQ_MASK,
+                           (NG_AT86RF2XX_IRQ_STATUS_MASK__RX_START |
+                            NG_AT86RF2XX_IRQ_STATUS_MASK__TRX_END));
+    /* go into RX state */
+    ng_at86rf2xx_set_state(dev, NG_AT86RF2XX_STATE_RX_AACK_ON);
+
+    DEBUG("ng_at86rf2xx_reset(): reset complete.\n");
+}
+
+bool ng_at86rf2xx_cca(ng_at86rf2xx_t *dev)
+{
+    uint8_t tmp;
+    uint8_t status;
+
+    /* trigger CCA measurment */
+    tmp = ng_at86rf2xx_reg_read(dev, NG_AT86RF2XX_REG__PHY_CC_CCA);
+    tmp &= NG_AT86RF2XX_PHY_CC_CCA_MASK__CCA_REQUEST;
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__PHY_CC_CCA, tmp);
+    /* wait for result to be ready */
+    do {
+        status = ng_at86rf2xx_reg_read(dev, NG_AT86RF2XX_REG__TRX_STATUS);
+    } while (!(status & NG_AT86RF2XX_TRX_STATUS_MASK__CCA_DONE));
+    /* return according to measurement */
+    if (status & NG_AT86RF2XX_TRX_STATUS_MASK__CCA_STATUS) {
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+size_t ng_at86rf2xx_send(ng_at86rf2xx_t *dev, uint8_t *data, size_t len)
+{
+    /* check data length */
+    if (len > NG_AT86RF2XX_MAX_PKT_LENGTH) {
+        DEBUG("[ng_at86rf2xx] Error: data to send exceeds max packet size\n");
+        return 0;
+    }
+    ng_at86rf2xx_tx_prepare(dev);
+    ng_at86rf2xx_tx_load(dev, data, len, 0);
+    ng_at86rf2xx_tx_exec(dev);
+    return len;
+}
+
+void ng_at86rf2xx_tx_prepare(ng_at86rf2xx_t *dev)
+{
+    uint8_t state;
+
+    /* make sure ongoing transmissions are finished */
+    do {
+        state = ng_at86rf2xx_get_state(dev);
+    }
+    while (state == NG_AT86RF2XX_STATE_BUSY_RX_AACK);
+    dev->idle_state = state;
+    ng_at86rf2xx_set_state(dev, NG_AT86RF2XX_STATE_TX_ARET_ON);
+    dev->frame_len = NG_IEEE802154_FCS_LEN;
+}
+
+size_t ng_at86rf2xx_tx_load(ng_at86rf2xx_t *dev, uint8_t *data,
+                            size_t len, size_t offset)
+{
+    dev->frame_len += (uint8_t)len;
+    ng_at86rf2xx_sram_write(dev, offset + 1, data, len);
+    return offset + len;
+}
+
+void ng_at86rf2xx_tx_exec(ng_at86rf2xx_t *dev)
+{
+    /* write frame length field in FIFO */
+    ng_at86rf2xx_sram_write(dev, 0, &(dev->frame_len), 1);
+    /* trigger sending of pre-loaded frame */
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__TRX_STATE,
+                           NG_AT86RF2XX_TRX_STATE__TX_START);
+    if (dev->event_cb && (dev->options & NG_AT86RF2XX_OPT_TELL_TX_START)) {
+        dev->event_cb(NETDEV_EVENT_TX_STARTED, NULL);
+    }
+}
+
+size_t ng_at86rf2xx_rx_len(ng_at86rf2xx_t *dev)
+{
+    uint8_t res;
+    ng_at86rf2xx_fb_read(dev, &res, 1);
+    return (size_t)(res - 2);           /* extract the PHR and LQI field */
```

Comment:
I don't understand `res - 2` in this context of your comment. Refering to Figure 38-4 on page 981, PHR (here = `res`) is the length of `Payload + FCS`, maybe you extract the checksum (=FCS) instead?

I'm investigating the strange LQI values and came here, because the return value is used to address the LQI field later on. So I'm trying to understand what's happening here.